controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -464,18 +458,44 @@ func (r *StorageConsumerReconciler) reconcileCephRadosNamespace(
 		}
 		rns.Namespace = r.namespace
 
-		if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rns, func() error {
-			if err := controllerutil.SetControllerReference(r.storageConsumer, rns, r.Scheme); err != nil {
-				return err
+		deleteRns := false
+		if !bp.DeletionTimestamp.IsZero() && bp.Status != nil {
+			for idx := range bp.Status.Conditions {
+				condition := &bp.Status.Conditions[idx]
+				// block pool waiting for rns deletion
+				if condition.Type == rookCephv1.ConditionDeletionIsBlocked &&
+					condition.Reason == rookCephv1.ObjectHasDependentsReason &&
+					condition.Status == corev1.ConditionTrue {
+					deleteRns = true
+				}
```

Comment:
Can you please explain why do we need this? This condition is not a blocker for our RNS deletion

Reply:
yes, not a blocker and at the same time, we do need to know when to delete the rns, ie, no blocked on images but only on rns cr.

Reply:
done.